src/main/kotlin/cacophony/grammar/syntaxtree/Expression.kt

```diff
@@ -0,0 +1,229 @@
+        type: Type,
+    ) : Expression(range)
+
+    class Literal(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class Keyword(
+        range: Pair<Location, Location>,
+    ) : Expression(range) {
+        class KeywordReturn(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+
+        class KeywordBreak(
+            range: Pair<Location, Location>,
+        ) : Expression(range)
+    }
+
+    // expression in parenthesis
+    class NestedExpression(
+        range: Pair<Location, Location>,
+        val expression: Expression,
+    ) : Expression(range)
+
+    // series of expressions separated by semicolons, returning value of the last one
+    // we agreed that (x; y;) returns Unit - should we create artificial class for empty expression after the last semicolon,
+    // or will we handle it in a different way?
+    class SubsequentExpressions(
+        range: Pair<Location, Location>,
+        vararg val expressions: Expression,
+    ) : Expression(range)
+
+    sealed class Statement(
+        range: Pair<Location, Location>,
+    ) : Expression(range) {
+        class IfStatement(
+            range: Pair<Location, Location>,
+            val testExpression: Expression,
+            val doExpression: Expression,
+        ) : Statement(range)
+
+        class IfElseStatement(
+            range: Pair<Location, Location>,
+            val testExpression: Expression,
+            val doExpression: Expression,
+            val elseExpression: Expression,
+        ) : Statement(range)
+
+        class WhileStatement(
+            range: Pair<Location, Location>,
+            val testExpression: Expression,
+            val doExpression: Expression,
+        ) : Statement(range)
+    }
+
+    class FunctionDefinition(
+        range: Pair<Location, Location>,
+        val functionName: VariableDeclaration,
+        vararg val arguments: VariableDeclaration,
+        val returnType: Type,
+    ) : Expression(range)
+
+    class FunctionCall(
+        range: Pair<Location, Location>,
+        val functionName: Variable,
+        vararg val arguments: Variable,
+    ) : Expression(range)
+
+    sealed class Operator(
+        range: Pair<Location, Location>,
+    ) : Expression(range) {
+        sealed class UnaryOperator(
+            range: Pair<Location, Location>,
+            val expression: Expression,
+        ) : Operator(range) {
+            class NegationOperator(
+                range: Pair<Location, Location>,
+                expression: Expression,
+            ) : UnaryOperator(range, expression)
+
+            class UnaryMinusOperator(
+                range: Pair<Location, Location>,
+                expression: Expression,
+            ) : UnaryOperator(range, expression)
+        }
+
+        sealed class BinaryOperator(
+            range: Pair<Location, Location>,
+            val expression1: Expression,
+            val expression2: Expression,
```

Comment:
all binary operators we have are infix, so i'd call these fields `lhs` and `rhs`

Reply:
Will change it